handlers/sf-billing-account-remover/src/main/scala/com/gu/sf_billing_account_remover/BillingAccountRemover.scala

```
@@ -0,0 +1,350 @@
+
+    parsedResponse match {
+      case Right(dr) =>
+        if (dr.Success) {
+          None
+        } else {
+          Some(
+            accountToDelete.copy(
+              ErrorCode = Some(dr.Errors(0).Code),
+              ErrorMessage = Some(dr.Errors(0).Message)
+            )
+          )
+        }
+      case Left(ex) => {
+        Some(accountToDelete.copy(ErrorMessage = Some(ex.toString)))
+      }
+    }
+
+  }
+
+  def insertSfErrorRecs(sfAuthDetails: SfAuthDetails,
+                        jsonBody: String): Either[Throwable, String] = {
+
+    val response =
+      try {
+        Right(
+          Http(
+            s"${sfAuthDetails.instance_url}/services/data/v45.0/composite/sobjects"
+          ).header("Authorization", s"Bearer ${sfAuthDetails.access_token}")
+            .header("Content-Type", "application/json")
+            .put(jsonBody)
+            .method("POST")
+            .asString
+            .body
+        )
+      } catch {
+        case ex: Throwable => Left(ex)
+      }
+    response
+  }
+
+  def updateSfBillingAccs(sfAuthDetails: SfAuthDetails,
+                          jsonBody: String): Either[Throwable, String] = {
+
+    val response =
+      try {
+        Right(
+          Http(
+            s"${sfAuthDetails.instance_url}/services/data/v45.0/composite/sobjects"
+          ).header("Authorization", s"Bearer ${sfAuthDetails.access_token}")
+            .header("Content-Type", "application/json")
+            .put(jsonBody)
+            .method("PATCH")
+            .asString
+            .body
+        )
+      } catch {
+        case ex: Throwable => { Left(ex) }
+      }
+    response
+  }
+
+  def updateZuoraBillingAcc(billingAccountForRemovalAsJson: String,
+                            zuoraBillingAccountId: String) = {
+    Http(
+      s"https://rest.apisandbox.zuora.com/v1/object/account/$zuoraBillingAccountId"
+    ).header("apiAccessKeyId", System.getenv("apiAccessKeyId"))
```

Better to put `apiAccessKeyId` in `Config` and pass `Config` through to method.  That way the program will fail fast if it doesn't have the value in its environment.

---

Good shout. This is done.